lib/entities/movie.dart

```diff
@@ -1,26 +1,68 @@
 class Movie {
+  final num voteCount;
+  final bool video;
+  final num popularity;
+  final String originalLanguage;
+  final num id;
+  final bool adult;
   final String title;
   final String originalTitle;
   final List<String> genres;
-  final String poster;
-  final String backdrop;
+  final String posterPath;
+  final String backdropPath;
   final String releaseDate;
-  final String voteAverage;
+  final num voteAverage;
   final String overview;
 
   const Movie(
+    this.voteCount,
+    this.video,
+    this.popularity,
+    this.originalLanguage,
+    this.id,
+    this.adult,
     this.title,
     this.originalTitle,
     this.genres,
-    this.poster,
-    this.backdrop,
+    this.posterPath,
+    this.backdropPath,
     this.releaseDate,
     this.voteAverage,
     this.overview,
   );
 
+  Movie.fromJson(Map<String, dynamic> json, List<dynamic> genresList)
+      : voteCount = json['vote_count'],
+        video = json['video'],
+        popularity = json['popularity'],
+        originalLanguage = json['original_language'],
+        id = json['id'],
+        adult = json['adult'],
+        title = json['title'],
+        originalTitle = json['original_title'],
+        posterPath = json['poster_path'],
+        genres = [],
+        backdropPath = json['backdrop_path'],
+        releaseDate = json['release_date'],
+        voteAverage = json['vote_average'],
+        overview = json['overview'] {
+    for (var id in json['genre_ids']) {
+      for (var genreItem in genresList) {
+        if (genreItem['id'] == id) {
+          genres.add(genreItem['name']);
+        }
+      }
+    }
```

Comment:
Instead of early mapping the genres by name, you could leave it directly as they come with the ids, then in the UI where you actually need to see the names there if you can do this kind of mapping where you get the name by id